internal/incident/incident.go

```diff
@@ -533,6 +481,74 @@ func (i *Incident) RestoreEscalationStateRules(states []*EscalationState) {
 	}
 }
 
+// getEscalationRecipientsChannel returns the configured channel of all the provided escalation recipients.
+func (i *Incident) getEscalationRecipientsChannel(escalation *rule.Escalation, t time.Time) recipientsChannelType {
+	if i.Recipients == nil {
+		i.Recipients = make(map[recipient.Key]*RecipientState)
+	}
+
+	managed := i.HasManager()
+	contactChannels := make(recipientsChannelType)
+	for _, escalationRecipient := range escalation.Recipients {
+		state := i.Recipients[escalationRecipient.Key]
+		if state == nil {
+			continue
+		}
+
+		if !managed || state.Role > RoleRecipient {
+			for _, c := range escalationRecipient.Recipient.GetContactsAt(t) {
+				if contactChannels[c] == nil {
+					contactChannels[c] = make(map[string]struct{})
+				}
+				if escalationRecipient.ChannelType.Valid {
+					contactChannels[c][escalationRecipient.ChannelType.String] = struct{}{}
+				} else {
+					contactChannels[c][c.DefaultChannel] = struct{}{}
+				}
+			}
+		}
+	}
+
+	return contactChannels
+}
+
+// getRecipientsChannel returns all the configured channels of the current incident and escalation recipients.
+func (i *Incident) getRecipientsChannel(t time.Time) recipientsChannelType {
+	if i.Recipients == nil {
+		i.Recipients = make(map[recipient.Key]*RecipientState)
+	}
```

Comment:
I don't see why this function initializes `i.Recipients` given that it never writes to that map anyways.